pkg/controller/apmserver/pod.go

```diff
@@ -120,7 +120,7 @@ func newPodSpec(as *v1alpha1.ApmServer, p PodSpecParams) corev1.PodTemplateSpec
 
 	if p.keystoreResources != nil {
 		dataVolume := keystore.DataVolume(
-			strings.ToLower(as.Kind()),
+			strings.ToLower(as.GetObjectKind().GroupVersionKind().Kind),
```

Comment:
can we just use `as.Kind` here? or is there something I am missing

Reply:
you're right, we only need to use GetObjectKind().GroupVersionKind().Kind when we reference the `HasKeystore` without having the associated type. That's because `Kind` is an attribute from `TypeMeta` from `k8s.io/apimachinery/pkg/apis/meta/v1`